presto-hive/src/test/java/com/facebook/presto/hive/TestHiveLogicalPlanner.java

```diff
@@ -1132,12 +1150,18 @@ public void testMaterializedViewOptimizationFullyMaterialized()
             assertUpdate(format("REFRESH MATERIALIZED VIEW %s WHERE ds = '2020-01-01'", view), 255);
             assertUpdate(format("REFRESH MATERIALIZED VIEW %s WHERE ds = '2019-01-02'", view), 14745);
 
-            String viewQuery = format("SELECT orderkey from %s where orderkey <  10000 ORDER BY orderkey", view);
-            String baseQuery = format("SELECT orderkey from %s where orderkey <  10000 ORDER BY orderkey", table);
+            String viewQuery = format("SELECT orderkey from %s where orderkey < 10000 ORDER BY orderkey", view);
+            String baseQuery = format("SELECT orderkey from %s where orderkey < 10000 ORDER BY orderkey", table);
 
             MaterializedResult viewTable = computeActual(viewQuery);
             MaterializedResult baseTable = computeActual(baseQuery);
             assertEquals(viewTable, baseTable);
+
+            assertPlan(getSession(), viewQuery, anyTree(
```

Comment:
Lets also check if there is no base table scan node present.

Reply:
Thanks for pointing this out, I believe in this case making sure no base table scan node is present is very important too. But after checking the code again, I think this `assertPlan` already guarantees thats.

Taking the following query plan for example (`viewQuery` in `testMaterializedViewOptimizationWithClause()`)
```
- Output[orderkey, orderpriority, ds]
    - RemoteStreamingMerge[orderkey_72 ASC_NULLS_LAST]
        - LocalMerge[orderkey_72 ASC_NULLS_LAST]
            - PartialSort[orderkey_72 ASC_NULLS_LAST]
                - RemoteStreamingExchange[REPARTITION]
                    - RemoteStreamingExchange[REPARTITION]
                        - ScanFilter[schemaName=tpch, tableName=test_orders_partitioned_with_clause]
                        - ScanFilter[schemaName=tpch, tableName=test_view_orders_partitioned_with_clause]
```

Only the assert with exact both scan nodes can pass:
```
assertPlan(getSession(), viewQuery, anyTree(
        anyTree(PlanMatchPattern.tableScan(table)),
        anyTree(PlanMatchPattern.tableScan(view))));

// ✅ Test passed
```

Missing any scan node, the assert will fail:
```
assertPlan(getSession(), viewQuery, anyTree(
        anyTree(PlanMatchPattern.tableScan(view))));

// ❌ java.lang.AssertionError: Plan does not match
```
```
assertPlan(getSession(), viewQuery, anyTree(
        PlanMatchPattern.tableScan(table)));

// ❌ java.lang.AssertionError: Plan does not match
```

Similarly, in our case, I think currently the `assertPlan` can make sure there is only a scan node on the view and no other table scan node present.